### ISSUE TITLE
BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#174]
## Abstract & Introduction
* BERT (Bidirectional Encoder Representations from Transformers) 제안.
* pre-trained BERT model --> output 레이어 하나만 더 붙여 fine-tuning시키면 다양한 문제에 우수한 성능으로 사용 가능.
* 1. 기존 feature-based approach인 ELMo 같은 경우에는 pre-trained representations을 추가적인 피쳐들로 포함하는 task-specific architectures를 사용함.
* 2. 기존 fine-tuning approach인 GPT는 minimal task-specific parameters를 도입 & pre-
trained parameter들을 fine-tuning해서 downstream tasks에 학습시킴.
* --> current techniques restrict the power of the pre-trained representations, especially for the fine-tuning approaches
* --> standard language models are unidirectional -> limits the choice of architectures that can be used during pre-training
* the importance of bidirectional pre-training for language representations
* pre-trained representations reduce the need for many heavily-engineered task-specific architectures

## BERT
* 두 개의 단계로 나뉨; pre-training & fine-tuning
  * pre-training: the model is trained on unlabeled data over different pre-training tasks
  * fine-tuning: the BERT model is first initialized with the pre-trained parameters, and all of the parameters are fine-tuned using labeled data from the downstream tasks
* pre-training과 fine-tuning 과정 (fine-tuning은 아래 그림에 나온 것 말고도 다양하게 수행 가능!)
  <image src="https://user-images.githubusercontent.com/67627471/135635036-708df772-327c-410a-9775-4ceff75e05c3.png" width="90%" height="90%" />
* 2가지 모델
  * BERT_BASE: L=12, H=768, A=12, Total Parameters=110M
  * BERT_LARGE: L=24, H=1024, A=16, Total Parameters=340M
* GPT는 모든 token -> only attend to context to its left
* BERT는 bidirectional self-attention 사용함.
* input에 single sentence / a pair of sentences 모두 가능.
* special tokens
  * 첫 번째 토큰은 항상 [CLS] 토큰.
  * [SEP] 토큰으로 문장을 나눔.
  <image src="https://user-images.githubusercontent.com/67627471/135637333-489c1104-4f46-4e33-907e-1f32df6a59b9.png" width="60%" height="60%" />
* Masked LM
  * bidirectional conditioning would allow each word to indirectly “see itself”
  * --> the model could trivially predict the target word in a multi-layered context
  * mask some percentage of the input tokens at random -> predict those masked tokens
  * mask 15% of all WordPiece tokens in each sequence at random
  * fine-tuning 시 이상 없도록  i번째 token이 선택되면.
    * (1) the [MASK] token 80% of the time
    * (2) a random token 10% of the time
    * (3) the unchanged i-th token 10% of the time
    으로 바꿈.
* Next Sentence Prediction (NSP)
  * sentence relationships을 학습시키기 위해 binarized next sentence prediction task를 pre-trainig함.
  * 두 개의 문장을 주고 이어져 있는지를 학습.
    * 이때 50%는 실제 이어져 있는 문장 / 나머지 50%는 랜덤으로 이어붙인 문장임.

## Results
* GLUE 테스트 결과.
  <image src="https://user-images.githubusercontent.com/67627471/135637500-5c605b2f-a77d-415c-a5cd-556956744f5b.png" width="80%" height="80%" />
* SQuAD 1.1 결과.
  <image src="https://user-images.githubusercontent.com/67627471/135637950-190214ba-4979-4603-827b-0c08299b2a08.png" width="40%" height="40%" />

##
* BEHRT 논문 읽기 전에 공부했다면 이해가 훨씬 쉬웠을..
* 처음보는 평가 매트릭이 많았음. --> 찾아보기.
